src/ompl/geometric/planners/experience/src/ThunderRetrieveRepair.cpp

```diff
@@ -126,7 +131,10 @@ namespace ompl
         base::PlannerStatus ThunderRetrieveRepair::solve(const base::PlannerTerminationCondition &ptc)
         {
             bool solved = false;
-            double approxdif = std::numeric_limits<double>::infinity();
+            double approxdif {0.0};
+            bool returnApproxSol {pdef_->getReturnApproximateSolutions()};
+            if (returnApproxSol)
+                approxdif = std::numeric_limits<double>::infinity();
```

Comment:
needs comments explaining why this is happening

Reply:
addressed